packages/components/src/SidePanel/SidePanel.component.js

```diff
@@ -91,73 +93,75 @@ function SidePanel({
 	const expandLabel = t('SIDEPANEL_EXPAND', { defaultValue: 'Expand' });
 	const collapseTitle = t('SIDEPANEL_COLLAPSE', { defaultValue: 'Collapse' });
 	const toggleButtonTitle = docked ? expandLabel : collapseTitle;
-
+	const Components = Inject.getAll(getComponent, { Action });
 	return (
 		<nav className={navCSS} role="navigation">
+			{injected('preul')}
 			<ul className={listCSS}>
 				{dockable && (
 					<li className={theme['toggle-btn']} title={toggleButtonTitle}>
-						<Action
+						<Components.Action
 							id={id && `${id}-toggle-dock`}
-							className={theme.link}
 							bsStyle="link"
 							onClick={onToggleDock}
 							icon="talend-opener"
 							label=""
 						/>
 					</li>
 				)}
-				{actions.map(action => {
-					const a11y = {};
-					const extra = {};
-					const isSelected = isActionSelected(action);
+				{injected('beforeactions')}
+				{actions &&
+					actions.map(action => {
```

Comment:
It would be good to externalize the menu in a SidePanelMenu component to have only 
```
{actions ?
    (<SidePanelMenu actions= {actions} selected={selected} getComponent={getComponent} />) :
    injected('actions')
}
```
in SidePanel render